src/coreclr/jit/compiler.h

```diff
@@ -9756,7 +9756,7 @@ class Compiler
     //
     bool isSIMDTypeLocalAligned(unsigned varNum)
     {
-#if defined(FEATURE_SIMD) && ALIGN_SIMD_TYPES
+#if defined(FEATURE_SIMD) && ALIGN_SIMD_TYPES && !UNIX_X86_ABI
```

Comment:
We can do better here, but this is no worse than win-x86 today.

Reply:
Would it be better to not define `ALIGN_SIMD_TYPES` for `UNIX_X86_ABI`? Makes the workaround more local.

Reply:
Either way would work and I am happy to change it if there's a preference. I chose this because it's closer to the code path that win-x86 takes today (by the virtue of having `STACK_ALIGN == 4`) but this is really only x86 usage of `ALIGN_SIMD_TYPES` so it wouldn't really matter. 

Ideally we want to do a similar thing as what x64 does in this branch and depend on the 16-byte stack alignment. We would just need to modify it to account for the pointer size being 4 bytes and not 8.

Reply:
ok